pkg/admissioncontroller/webhook/auth/shoot/authorizer.go

```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package shoot
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	auth "k8s.io/apiserver/pkg/authorization/authorizer"
+
+	shootidentity "github.com/gardener/gardener/pkg/admissioncontroller/gardenletidentity/shoot"
+	authwebhook "github.com/gardener/gardener/pkg/admissioncontroller/webhook/auth"
+	"github.com/gardener/gardener/pkg/utils/graph"
+	authorizerwebhook "github.com/gardener/gardener/pkg/webhook/authorizer"
+)
+
+// NewAuthorizer returns a new authorizer for requests from gardenlets running in autonomous shoots.
+func NewAuthorizer(logger logr.Logger, graph graph.Interface, authorizeWithSelectors authorizerwebhook.WithSelectorsChecker) *authorizer {
+	return &authorizer{
+		logger:                 logger,
+		graph:                  graph,
+		authorizeWithSelectors: authorizeWithSelectors,
+	}
+}
+
+type authorizer struct {
+	logger                 logr.Logger
+	graph                  graph.Interface
+	authorizeWithSelectors authorizerwebhook.WithSelectorsChecker
+}
+
+var _ = auth.Authorizer(&authorizer{})
+
+func (a *authorizer) Authorize(_ context.Context, attrs auth.Attributes) (auth.Decision, string, error) {
+	shootNamespace, shootName, isAutonomousShoot, userType := shootidentity.FromUserInfoInterface(attrs.GetUser())
+	if !isAutonomousShoot {
+		return auth.DecisionNoOpinion, "", nil
+	}
+
+	var (
+		log               = a.logger.WithValues("shootNamespace", shootNamespace, "shootName", shootName, "attributes", fmt.Sprintf("%#v", attrs), "userType", userType)
+		requestAuthorizer = &authwebhook.RequestAuthorizer{
+			Log:                    a.logger,
```

Comment:
I am not so sure about this, but shouldn't it be?
```suggestion
			Log:                    log,
```